src/core/service.c

```diff
@@ -903,39 +919,64 @@ static int service_load(Unit *u) {
         return service_verify(s);
 }
 
+static void service_dump_fd(int fd, const char *fdname, const char *header, FILE *f, const char *prefix) {
+        _cleanup_free_ char *path = NULL;
+        struct stat st;
+        int flags;
+
+        if (fstat(fd, &st) < 0) {
+                log_debug_errno(errno, "Failed to stat fdstore entry: %m");
+                return;
+        }
```

Comment:
Not a big matter, but please propagate the error, then ignore the error code in the caller side.